crates/bevy_diagnostic/src/lib.rs

```diff
@@ -15,10 +16,47 @@ pub struct DiagnosticsPlugin;
 
 impl Plugin for DiagnosticsPlugin {
     fn build(&self, app: &mut App) {
-        app.init_resource::<Diagnostics>();
+        app.init_resource::<Diagnostics>()
+            .add_startup_system(log_system_info);
     }
 }
 
 /// The width which diagnostic names will be printed as
 /// Plugin names should not be longer than this value
 pub const MAX_DIAGNOSTIC_NAME_WIDTH: usize = 32;
+
+#[derive(Debug)]
+// This is required because the Debug trait doesn't detect it's used when it's only used in a print :(
+#[allow(dead_code)]
+struct SystemInfo {
+    os: String,
+    kernel: String,
+    cpu: String,
+    core_count: String,
+    memory: String,
+}
+
+fn log_system_info() {
+    use sysinfo::{CpuExt, SystemExt};
+
+    let mut sys = sysinfo::System::new();
+    sys.refresh_cpu();
+    sys.refresh_memory();
+
+    let info = SystemInfo {
+        os: sys
+            .long_os_version()
+            .unwrap_or_else(|| String::from("not available")),
+        kernel: sys
+            .kernel_version()
+            .unwrap_or_else(|| String::from("not available")),
+        cpu: sys.global_cpu_info().brand().trim().to_string(),
+        core_count: sys
+            .physical_core_count()
+            .map(|x| x.to_string())
+            .unwrap_or_else(|| String::from("not available")),
+        memory: format!("{} KB", sys.total_memory()),
```

Comment:
Any chance of getting this is GB instead? Or at least MB

Reply:
Not directly, but I can easily just divide it by 0.000001 to get the amount in GB

Reply:
Either that or formatting it so it has dividers making it obvious. I prefer space dividers

Reply:
I decided to convert it to gibibyte instead since it's probably what most people expect since it's the unit used in most marketing material when buying ram sticks

Reply:
Well, even that's not completely true since they have a tendency to also round that value

Reply:
Actually, I got that backwards, windows reports it in GiB, but manufacturers use GB